pkg/iam/iam.go

```diff
@@ -182,13 +193,13 @@ func (s *IAMService) generateRoute53RoleParams(roleTypeToReconcile string, awsAc
 			return Route53RoleParams{}, err
 		}
 
-		kiamRoleARN = *o.Role.Arn
+		principalRoleARN = *o.Role.Arn
 	}
 
 	if s.roleType == KIAMRole {
 		params := Route53RoleParams{
 			EC2ServiceDomain: ec2ServiceDomain(s.region),
-			KIAMRoleARN:      kiamRoleARN,
+			PrincipalRoleARN: principalRoleARN,
```

Comment:
KIAM role arn was misleading as it was IRSA role anyway, but to make it more general I renamed it to `PrincipalRoleARN` which correlates to AWS IAM terminology